src/NRules/NRules.Fluent/Expressions/FilterExpression.cs

```diff
@@ -34,7 +44,45 @@ public IFilterExpression Where(params Expression<Func<bool>>[] predicates)
                 var expression = predicate.DslExpression(filters.Declarations);
                 filters.Filter(FilterType.Predicate, expression);
             }
+
             return this;
         }
+
+        private static Expression<Func<object>> InternalizeComparer<T>(Expression<Func<T>> keySelector, Expression<Func<IEqualityComparer<T>>> comparer) where T : class
```

Comment:
This seems pretty hacky. I don't think the equality wrapper should be part of the engine, especially given the restriction on the type argument to be a reference type

Reply:
Agreed. I'll implement this as an extension in my application then.